### PR TITLE
ARMeilleure: A32: Implement SHSUB8 and UHSUB8

### PR DESCRIPTION
**Pull request by merryhime**
Based on [this commit](https://github.com/merryhime/dynarmic/commit/0ff8c375af74a0d20aee1e383e4ed509808a0314).

### CODE REVIEW

File: ARMeilleure/Instructions/InstEmitAlu32.cs [hunk 1]
```diff
@@ -659,6 +669,42 @@ private static void EmitDiv(ArmEmitterContext context, bool unsigned)
             context.MarkLabel(lblEnd);
         }
 
+        private static void EmitHsub(ArmEmitterContext context, bool unsigned)
```
Comment (gdkchan) [inline ARMeilleure/Instructions/InstEmitAlu32.cs:672]: I'd call it `EmitHsub8`.

File: Ryujinx.Tests/Cpu/CpuTestAlu32.cs [hunk 2]
```diff
@@ -77,6 +77,25 @@ private static uint[] _Lsr_Lsl_Asr_Ror_()
...
 
+        [Test, Pairwise]
+        public void Shsub8([Values(0u, 0xdu)] uint rd,
+                           [Values(1u)] uint rm,
+                           [Values(2u)] uint rn,
+                           [Random(RndCnt)] uint w0,
+                           [Random(RndCnt)] uint w1,
+                           [Random(RndCnt)] uint w2)
+        {
+            uint opcode = 0xE6300FF0u; //SHSUB8 R0, R0, R0
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestAlu32.cs:88]:
```suggestion
            uint opcode = 0xE6300FF0u; // SHSUB8 R0, R0, R0
```

File: Ryujinx.Tests/Cpu/CpuTestAlu32.cs [hunk 3]
```diff
@@ -130,6 +149,25 @@ private static uint[] _Lsr_Lsl_Asr_Ror_()
...
+
+        [Test, Pairwise]
+        public void Uhsub8([Values(0u, 0xdu)] uint rd,
+                           [Values(1u)] uint rm,
+                           [Values(2u)] uint rn,
+                           [Random(RndCnt)] uint w0,
+                           [Random(RndCnt)] uint w1,
+                           [Random(RndCnt)] uint w2)
+        {
+            uint opcode = 0xE6700FF0u; //UHSUB8 R0, R0, R0
```
Comment (gdkchan) [inline Ryujinx.Tests/Cpu/CpuTestAlu32.cs:161]:
```suggestion
            uint opcode = 0xE6700FF0u; // UHSUB8 R0, R0, R0
```

**Review by gdkchan — APPROVED** [review]
lgtm, thanks.

**Review by AcK77 — APPROVED** [review]

**Pull request merged**